File: app/src/main/java/me/shreyasr/chatse/chat/ChatFragmentPagerAdapter.java
package me.shreyasr.chatse.chat;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class ChatFragmentPagerAdapter extends FragmentPagerAdapter {

    private List<ChatFragment> chatFragments = new ArrayList<>();

    public ChatFragmentPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public void addFragment(ChatFragment fragment) {
        chatFragments.add(fragment);
        notifyDataSetChanged();
    }

    public void removeFragment(ChatFragment fragment) {
        chatFragments.remove(fragment);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return chatFragments.size();
    }

    @Override
    public Fragment getItem(int pos) {
        return chatFragments.get(pos);
    }

    @Override
    public String getPageTitle(int pos) {
        return chatFragments.get(pos).getPageTitle();
    }
}

File: app/src/main/java/me/shreyasr/chatse/chat/ChatFragmentPagerAdapter.kt
package me.shreyasr.chatse.chat

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.*

class ChatFragmentPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val chatFragments = ArrayList<ChatFragment>()

    fun addFragment(fragment: ChatFragment) {
        chatFragments.add(fragment)
        notifyDataSetChanged()
    }

    fun removeFragment(fragment: ChatFragment) {
        chatFragments.remove(fragment)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return chatFragments.size
    }

    override fun getItem(pos: Int): Fragment {
        return chatFragments[pos]
    }

    override fun getPageTitle(pos: Int): String {
        return chatFragments[pos].pageTitle
    }
}
